### PR TITLE
HPCC-18114 Unicode implementation of removeSuffix

### PR DESCRIPTION
**Pull request by ghalliday**
- Adds test cases in ecllibrary/teststd/uni/TestRemoveSuffix.ecl
- Adds documentation for removeSuffix in ecllibrary/std/uni/uni.ecl
- Adds code for RemoveSuffix in plugins/unicodelib/unicodelib.cpp and .hpp

Signed-off-by: David Skaff <david.skaff@lexisnexisrisk.com>
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x] Performance
  - [ ] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review.
The second commit cleans up various problems with trimming unicode strings (were previously trimmed from the start of some strings, and not trimming some others.)

File: ecllibrary/teststd/uni/TestEndsWith.ecl [hunk 1]
```diff
@@ -36,7 +36,10 @@ EXPORT TestEndsWith := MODULE
     //Check action on a string containing different variations/combinations of numbers and other characters.
     EXPORT Test24 := ASSERT(Uni.EndsWith('123abc 23.6 abc123','23.6 abc123','') = TRUE);
     //Test other space characters (< 0x20).
-    EXPORT Test26 := ASSERT(Uni.EndsWith('\nt\t','t','') = FALSE);
+    //It appears the unicode trim semantics are to remove whitespace characters 0 that was not the intention, it
```
Comment (richardkchapman) [inline ecllibrary/teststd/uni/TestEndsWith.ecl]: Is the '0' in this comment a typo for - ?

Not sure that the fact that TRIM(unicode) is wrong necessarily means that Uni.EndsWith() should be wrong?

**Comment by richardkchapman** [comment]
@ghalliday One minor comment and one question

File: ecllibrary/teststd/uni/TestEndsWith.ecl (hunk 1 above)
Comment (ghalliday) [reply ecllibrary/teststd/uni/TestEndsWith.ecl]: yes 0 must have replaced the text.
The trim function is also used by the edit distance - at least it is consistent.  @richardkchapman Should I change it, or leave it as it is?

Comment (ghalliday) [reply ecllibrary/teststd/uni/TestEndsWith.ecl]: conclusion is that trim should not remove arbitrary whitespace.  That should be fixed, and another function added which does trim whitespace (and is also available for strings).

**Comment by richardkchapman** [comment]
@ghalliday I think this is back in your court at present

**Comment by ghalliday** [comment]
@richardkchapman rebased, commits squashed, and test cases updated to reflect new implicit trim semantics.

**Pull request merged**